framework/src/fvkernels/FVScalarLagrangeMultiplier.C

```diff
@@ -0,0 +1,84 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVScalarLagrangeMultiplier.h"
+
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+
+#include "MooseVariableScalar.h"
+#include "MooseVariableFV.h"
+#include "Assembly.h"
+
+registerMooseObject("MooseApp", FVScalarLagrangeMultiplier);
+
+InputParameters
+FVScalarLagrangeMultiplier::validParams()
+{
+  InputParameters params = FVElementalKernel::validParams();
+  params.addClassDescription("This class is used to enforce integral of phi = V_0 with a "
+                             "Lagrange multiplier approach.");
```

Comment:
It can be whatever you want to average value to be. We can just set it to zero for now and change it I suppose if a user wants to change the average value

Reply:
I went ahead and made V0 = 0

Reply:
But what if I want an average that makes physical sense? Like to have the primary loop pressure at the right average 
Sorry I didnt reply earlier.

Reply:
Is this loop fully enclosed without inflow or outflow boundaries?

Reply:
:cry:

Reply:
Sorry. I ll get you a beer when you come over

Reply:
Ok @GiudGiud your request hopefully resolved in 25984d706d

Reply:
> Sorry. I ll get you a beer when you come over

Haha I like that!